CONTRIBUTING.md

```diff
@@ -1,43 +1,119 @@
-## Development environment
+## Set up the development environment
 
 We recommend uv for setting up your local development environment:
 
 - [uv snap](https://snapcraft.io/astral-uv)
 - [Official uv binary](https://docs.astral.sh/uv/getting-started/installation/)
 
-To set up an initial development environment:
+To set up a local copy of the repository for development:
 
-    git clone https://github.com/canonical/charmcraft.git
-    cd charmcraft
-    uv sync --all-extras
+```bash
+git clone https://github.com/canonical/charmcraft.git
+cd charmcraft
+uv sync --all-extras
```

Comment:
@lengau Since this is different from `uv sync --frozen`, should we branch for doc setup? Or should this in fact also be `make setup`?

Reply:
That should be `make setup`. Good catch!
```suggestion
make setup
```

Reply:
[b64695f](https://github.com/canonical/charmcraft/pull/2101/commits/b64695fde4a4d5d5166aedf24a9f053a7d4146ed)